assets/aml-benchmark/components/dataset-preprocessor/spec.yaml

```diff
@@ -0,0 +1,56 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+name:  dataset_preprocessor 
+display_name: Dataset Preprocessor
+description: Dataset Preprocessor
+version: 0.0.1
+is_deterministic: true
+
+inputs:
+  dataset: 
+    type: uri_file
+    description: >-
+      Path to load the dataset.
+    optional: False
+  template_input:
+    type: string
+    description: >-
+      A json dictionary where key is the name of the column enclosed in " " and associated dict value is presented using jinja template logic 
+      which will be used to extract respective value from the dataset. 
+      Example format:
+      {"<user_column_name>": {{key in the json file for this column}}, ....}. The processed output will be dumped to a 
+      jsonl file in this format: {"<user_column_name>": "<key in the json file for this column>", ....} 
+    optional: True
+  script_path:
+    type: uri_file
+    description: >-
+      Path to the custom preprocessor python script provided by user. If this is provided, then, this takes precedence.  
+    optional: True
+  encoder_config:
+    type: string
+    description: >-
+      Dictionary in json format that contains column/key name to encode associated with dict key `column_name` followed by  
```

Comment:
> Dictionary in json format that contains column/key name to encode associated with dict key

The sentence is confusing. Associated with what?

Reply:
Associated with dict key "column_name" ->  key in the encoder_config dictionary.

Reply:
How does it differ from template_input? In one of the tests, you have-
template_input-
```
'{"question":{{question}},"solution":{{answer.split("#### ")[0]}},"answer":{{answer.split("#### ")[-1]|string}}}'
```
encoder_config-
```
 '{"context":{{ctx|tojson}},"endings":{{endings|tojson}},"option1":{{endings[0]|tojson}}}'
```

What does template_input solve that encoder_config can't and vice-versa?

Reply:
@arun-rajora This ` '{"context":{{ctx|tojson}},"endings":{{endings|tojson}},"option1":{{endings[0]|tojson}}}'` is just a placeholder in yaml. The actual values to test are passed in test script test_dataset_preprocessor.py shown below : 

@pytest.mark.parametrize(
        "dataset_name, dataset,template_input,script_path, encoder_config",
        [
            (
                "gsm8k", INPUT_DATASET,
                """ {
                    "question":{{question}},
                    "solution":{{answer.split("#### ")[0]}},
                    "answer":{{answer.split("#### ")[-1]|string}}
                } """,
                None, None
            ),
            ("quac_org", INPUT_DATASET, None, os.path.join(CUSTOM_SCRIPT_PATH, "quac_textgen_babel.py"), None),
            (
                "mnli", INPUT_DATASET,
                """ {"premise":{{premise}}, "hypothesis":{{hypothesis}},"label":{{label|string}}} """,
                None, '{"column_name":"label", "0":"NEUTRAL", "1":"ENTAILMENT", "2":"CONTRADICTION"}'
            ),
        ],
    )

Reply:
Got it. Thanks!